File: backend/internal/models/params.go
package models

type Page struct {
	Limit  int
	Offset int
}

type Sort struct {
	Field string `json:"field"`
	Type  string `json:"type"`
}

type Filter struct {
	Field     string         `json:"field"`
	FieldType string         `json:"fieldType"`
	Values    []*FilterValue `json:"values"`
}
type FilterValue struct {
	CompareType string `json:"compareType"`
	Value       string `json:"value"`
}

File: frontend/src/app/types/params.ts
export interface IParams {
	page?: number
	size?: number
	sort?: ISort
	filters?: IFilter[]
}

export interface ISort {
	[x: string]: 'DESC' | 'ASC'
}

export type CompareTypes = 'con' | 'start' | 'end' | 'like' | 'in' | 'eq' | 'gte' | 'lte'
export interface IFilter {
	field: string
	fieldType: string
	compareType: CompareTypes
	value: string
}
